core/src/main/java/io/aiven/kafka/tieredstorage/RemoteStorageManager.java

```diff
@@ -105,14 +113,29 @@ public RemoteStorageManager() {
 
     // for testing
     RemoteStorageManager(final Time time) {
+        this.time = time;
+
         final JmxReporter reporter = new JmxReporter();
         metrics = new Metrics(
             new MetricConfig(), List.of(reporter), time,
             new KafkaMetricsContext("aiven.kafka.server.tieredstorage")
         );
         segmentCopyPerSec = metrics.sensor("segment-copy");
-        segmentCopyPerSec.add(
-            metrics.metricName("segment-copy-rate", "remote-storage-manager-metrics"), new Rate());
+        final String metricGroup = "remote-storage-manager-metrics";
+        segmentCopyPerSec.add(metrics.metricName("segment-copy-rate", metricGroup), new Rate());
+
+        segmentCopyTime = metrics.sensor("segment-copy-time");
+        segmentCopyTime.add(metrics.metricName("segment-copy-time-avg", metricGroup), new Avg());
+        segmentCopyTime.add(metrics.metricName("segment-copy-time-max", metricGroup), new Max());
+
+        segmentFetchPerSec = metrics.sensor("segment-fetch");
+        segmentFetchPerSec.add(metrics.metricName("segment-fetch-rate", metricGroup), new Rate());
+
+        segmentFetchToFirstByteTime = metrics.sensor("segment-fetch-to-first-byte-time");
+        segmentFetchToFirstByteTime.add(metrics.metricName("segment-fetch-to-first-byte-time-avg", metricGroup),
+            new Avg());
+        segmentFetchToFirstByteTime.add(metrics.metricName("segment-fetch-to-first-byte-time-max", metricGroup),
+            new Max());
```

Comment:
Could we add some comments (not sure if here or in README maybe?) about how this metric is going to be used? Others are pretty standard, but time to first byte seems to deliver some insights that are not obvious.

Reply:
I removed this one for now. Need more thinking of how to better implement it and next time I'll add a description